docs/content/sips/0xx-component-dependencies.md

```diff
@@ -0,0 +1,108 @@
+title = "SIP 0XX - Component Dependencies"
+template = "main"
+date = "2024-05-31T12:00:00Z"
+---
+Summary: This feature enables developers to specify and manage dependencies for Spin components, allowing the use of components from various languages and sources, either locally or remotely, with configurable inheritance of parent component configuration.
+
+Owner(s): [brian.hardock@fermyon.com](mailto:brian.hardock@fermyon.com)
+
+Created: May 31, 2024
+
+# Background
+This feature enables the polyglot re-use of components, allowing developers to specify how to use components written in various languages as libraries to fulfill dependencies in their Spin components.
+
+For instance, consider writing a Spin component in Go that depends on the `aws:client/s3` interface (expressed as an import in the compiled Go component, e.g., `import aws:client/s3`). To satisfy this dependency, you can use a componentized S3 client written in Rust that exports an instance of the `aws:client/s3` interface. Using the proposed syntax in this SIP, Spin developers can instruct Spin on how to compose these components together to achieve this.
```

Comment:
Do we allow for the possibility that the dependency has imports, e.g. callbacks?  Is that a "hie thee to `wac`" situation?

Reply:
I wasn't planning on placing any restrictions on the shape of dependencies. That being said, if a dependency contains imports that can not be satisfied by Spin the component will fail to instantiate.